### DISCUSSION TITLE
KNX: Add RGB and Dimming Control

### DISCUSSION THREAD

**Discussion by goofy79** [arendst/Tasmota#11733] [Ideas]
### PROBLEM DESCRIPTION
I think the KNX Settings for the RGB Module doesn't work ?
I have **Arilux LC01** Modul with the Tasmota Version 8.1.0.
In the KNX Settings, i can choose KNX RX 1 - KNX RX 5
But it doesn't matter what data i send to the Inputs - the Color and the brightness
won't change.

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [x] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [x] Searched the problem in the [docs](https://tasmota.github.io/docs/#/help/FAQ)
- [x] Searched the problem in the [forum](https://groups.google.com/d/forum/sonoffusers)
- [x] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [x] Device used (e.g., Sonoff Basic): Arilux LC01
- [x] Tasmota binary firmware version number used: 8.0.1
  - [x] Pre-compiled
- [x] Flashing tools used: nodemcu pyflasher
- [x] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```
  Configuration output here:

16:35:12 CMD: Backlog Template; Module; GPIO 255
16:35:12 RSL: RESULT = {"NAME":"Generic","GPIO":[255,255,255,255,255,255,255,255,255,255,255,255,255],"FLAG":15,"BASE":18}
16:35:12 RSL: RESULT = {"Module":{"37":"Arilux LC01"}}
16:35:12 RSL: RESULT = {"GPIO0":{"17":"Button1"},"GPIO1":{"0":"None"},"GPIO2":{"159":"ALux IrSel"},"GPIO3":{"0":"None"},"GPIO4":{"147":"ALux IrRcv"},"GPIO5":{"37":"PWM1"},"GPIO9":{"0":"None"},"GPIO10":{"0":"None"},"GPIO12":{"38":"PWM2"},"GPIO13":{"39":"PWM3"},"GPIO14":{"0":"None"},"GPIO15":{"0":"None"},"GPIO16":{"0":"None"}}

```

- [x] Provide the output of this command: `Status 0`:
```
  STATUS 0 output here:

16:37:37 CMD: Status 0
16:37:37 RSL: STATUS = {"Status":{"Module":37,"FriendlyName":["Tasmota"],"Topic":"ARILUX_LC01","ButtonTopic":"0","Power":1,"PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0}}
16:37:37 RSL: STATUS1 = {"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"tasmotas","OtaUrl":"http://thehackbox.org/tasmota/release/tasmota.bin","RestartReason":"External System","Uptime":"1T02:26:29","StartupUTC":"2020-01-03T13:11:08","Sleep":50,"CfgHolder":4617,"BootCount":6,"SaveCount":171,"SaveAddress":"F9000"}}
16:37:37 RSL: STATUS2 = {"StatusFWR":{"Version":"8.1.0.1(tasmota)","BuildDateTime":"2019.12.31 14:39:21","Boot":7,"Core":"2_6_1","SDK":"2.2.2-dev(38a443e)","Hardware":"ESP8285","CR":"366/699"}}
16:37:37 RSL: STATUS3 = {"StatusLOG":{"SerialLog":2,"WebLog":2,"MqttLog":0,"SysLog":0,"LogHost":"","LogPort":514,"SSId":["FRITZ!Box Fon WLAN 7170",""],"TelePeriod":300,"Resolution":"558180C0","SetOption":["0A008001","2805C8000100060000005A00000000000000","00000000","00000000"]}}
16:37:37 RSL: STATUS4 = {"StatusMEM":{"ProgramSize":595,"Free":408,"Heap":22,"ProgramFlashSize":1024,"FlashSize":1024,"FlashChipId":"144051","FlashMode":3,"Features":["00000407","9FDAE397","043683A0","22B617CD","01001BC0","00007881"],"Drivers":"1,2,3,4,5,6,7,8,9,10,11,12,16,18,19,20,21,22,24,26,29","Sensors":"1,2,3,4,5,6,7,8,9,10,14,15,17,18,20,22,26,34"}}
16:37:37 RSL: STATUS5 = {"StatusNET":{"Hostname":"ARILUX_LC01-6912","IPAddress":"172.16.0.32","Gateway":"172.16.0.1","Subnetmask":"255.255.255.0","DNSServer":"172.16.0.1","Mac":"D8:F1:5B:86:FB:00","Webserver":2,"WifiConfig":4,"WifiPower":17.0}}
16:37:37 RSL: STATUS7 = {"StatusTIM":{"UTC":"Sat Jan 04 15:37:37 2020","Local":"Sat Jan 04 16:37:37 2020","StartDST":"Sun Mar 29 02:00:00 2020","EndDST":"Sun Oct 25 03:00:00 2020","Timezone":"+01:00","Sunrise":"08:43","Sunset":"17:06"}}
16:37:37 RSL: STATUS10 = {"StatusSNS":{"Time":"2020-01-04T16:37:37"}}
16:37:37 RSL: STATUS11 = {"StatusSTS":{"Time":"2020-01-04T16:37:37","Uptime":"1T02:26:29","UptimeSec":95189,"Heap":22,"SleepMode":"Dynamic","Sleep":10,"LoadAvg":99,"MqttCount":0,"POWER":"AN","Dimmer":89,"Color":"00E38F","HSBColor":"158,100,89","Channel":[0,89,56],"Scheme":0,"Fade":"AUS","Speed":1,"LedTable":"AN","Wifi":{"AP":1,"SSId":"FRITZ!Box Fon WLAN 7170","BSSId":"08:96:D7:57:F7:08","Channel":1,"RSSI":30,"Signal":-85,"LinkCount":3,"Downtime":"0T00:00:10"}}}
```

### TO REPRODUCE
Set KNX Group Addresses with ETS - than set the Groupaddresses the receive Date an send any data to this Groupaddresses. 
My Groupaddresses are 
3/3/70 -> Output 1 On/Off **(work)**
3/3/71 -> KNX RX 1 _(won't work with any data)_
3/3/72 -> KNX RX 2 _(won't work with any data)_
3/3/73 -> KNX RX 3 _(won't work with any data)_
3/3/74 -> KNX RX 4 _(won't work with any data)_
3/3/75 -> KNX RX 5 _(won't work with any data)_

### EXPECTED BEHAVIOUR
I expected that, if i send a value between 0-255 the the KNX RX 1-3 the Color of the RGB Stripe will change. So KNX RX 1 for Red - KNX RX 2 for Green - KNX RX 3 for Blue - KNX RX 4 maybe for brightness and KNX RX 5 for the last control.
So i can change the color with a string linke 
0/0/255 for Blue
255/0/255 for Pink
and so on ...

I think the Values are there
`16:43:56 RSL: RESULT = {"POWER":"AN","Dimmer":89,"Color":"00E38F","HSBColor":"`**158,100,89**`","Channel":[0,89,56]}`
the only thing to do is, to set the received values from the KNX objects to the right variables, isn't it  ?

### SCREENSHOTS
![Bildschirmfoto 2020-01-04 um 16 57 58](https://user-images.githubusercontent.com/11212295/71768095-61357980-2f13-11ea-8395-53d83a3f3ef7.png)

![Bildschirmfoto 2020-01-04 um 16 59 36](https://user-images.githubusercontent.com/11212295/71768117-9e9a0700-2f13-11ea-9141-01cc09123d32.png)

![Bildschirmfoto 2020-01-04 um 16 45 41](https://user-images.githubusercontent.com/11212295/71767914-aeb0e700-2f11-11ea-8587-1ea56f40186a.png)

![Bildschirmfoto 2020-01-04 um 16 46 39](https://user-images.githubusercontent.com/11212295/71767943-fafc2700-2f11-11ea-84b1-5a1386621c36.png)

### ADDITIONAL CONTEXT
nothing at the moment

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by ascillato** [comment]
KnxRx slots are meant to be used with rules. See wiki and docs at KNX chapter on how to use knx rules triggers. If you send the color separated in channels, you should be able to set rules to use them and set the colors in Tasmota. 

If the command you are sending is the DPT for color, it is not supported by now.

**Comment by goofy79** [comment]
Ok - is this planed ?
i tried

```
rule1 1
rule1 on event#knxrx_val1 do Var1 %value% endon
rule1 + on event#knxrx_val2 do Var2 %value% endon
rule1 + on event#knxrx_val3 do Var3 %value% endon
rule1 + on event#knxrx_val5 do Dimmer %value% endon
rule1 + on Var1#State do Color1 %Var1%,%Var2%,%Var3% endon
rule1 + on Var2#State do Color1 %Var1%,%Var2%,%Var3% endon
rule1 + on Var3#State do Color1 %Var1%,%Var2%,%Var3% endon
```

```
18:38:52 KNX: Received from 3.3.71 Command Write: 41943.04 to KNX RX 1
18:38:52 RSL: stat/tasmota/RESULT = {"Event":"Done"}
18:38:53 RUL: EVENT#KNXRX_VAL1 performs "var1 41943.0"
18:38:53 RSL: stat/tasmota/RESULT = {"Var1":"41943.0"}
18:38:53 RUL: VAR1#STATE performs "Color1 41943.0,10.24,41943.0"
18:38:53 RSL: stat/tasmota/RESULT = {"POWER":"ON","Dimmer":84,"Color":"215,71,215","HSBColor":"300,95,84","Channel":[84,4,84]}
```
But i send for example the Value 100 to the Object and 41943 / 215 arrives ? 
If i send 10 - 10,96 (10) arrives / if i send 50 - 327.68 arrives ?

```
18:40:34 RUL: EVENT#KNXRX_VAL2 performs "var2 327.68"
18:40:34 RSL: stat/tasmota/RESULT = {"Var2":"327.68"}
18:40:34 RUL: VAR2#STATE performs "Color1 41943.0,327.68,41943.0"
18:40:34 RSL: stat/tasmota/RESULT = {"POWER":"ON","Dimmer":84,"Color":"215,71,215","HSBColor":"300,67,84","Channel":[84,28,84]}
```
Whats wrong with my rule ?
Thank you so much

**Comment by ascillato2** [comment]
Hi, yes, it is planned to be added. I have to find some time to do it.

About your rule. Your rule is ok. The issue is the DPT your KNX system is using to send that value that is not what Tasmota expects. Try to change the DPT. If you can't you will have to wait until Color DPT support is added to the Tasmota KNX driver.

**Comment by goofy79** [comment]
can you tell me what kind of DPT the Tasmota driver expected ?
Actual i send 5.005 Decimal (0-255)

**Comment by ascillato2** [comment]
Yes, sorry, Tasmota expects DPT 9 (2 bytes float) for recepted values. I have to add that information to the docs. Thanks.

**Comment by goofy79** [comment]
puh, that's heavy - to convert a 0-255 RGB value in KNX to a 2 Byte Float.
my Visu send, for example, a 1-Byte value per Color

![Bildschirmfoto 2020-01-05 um 19 00 31](https://user-images.githubusercontent.com/11212295/71783912-ac23bf80-2fed-11ea-8695-c45c08284b1d.png)

perfect will be, if you can set the datatype for receiving ^^

PS: if you need somebody to test - feel free to write 👍

**Comment by JohannesRegner** [comment]
Hi, i have the same problem! I'm using an Gira X1 KNX Visua Server, and tryed to change the color via GUI, but i can only switch on/off.
I think, the problem is the same as @goofy79  described ...

 if you need any support/testing, I can also help!

**Comment by githubtzzz** [comment]
Hi, any updates here?

**Comment by ascillato** [comment]
Hi, yes, it is planned to be added. I have to find some time to do it.

**Comment by githubtzzz** [comment]
Hi @ascillato, do you think you can find time for this?
Do you need any support?

**Comment by ascillato2** [comment]
@githubtzzz Please, give me some days. Thanks.

**Comment by ascillato2** [comment]
From #8084, also add Dimming control.

From #10333 also DPT Selection

**Comment by fluppie** [comment]
+1 I was also looking for this func. I can switch the LED strip on and off, but also want to regulate the brightness with the KNX switches.

**Comment by JackyKNX** [comment]
+1 more ;) Is there any chance to get it soon ?

**Comment by jpirnay** [comment]
+1, I do have some tasmota driven LED strips that would definitely benefit from this!

**Comment by edvandreas** [comment]
+1, I would also like to get this implemented

**Comment by Jaegerstefan** [comment]
+1 It would be really helpful to get this working.

**Comment by eldi0s** [comment]
Definitely +1

I wonder how DPT9 could make it while DPT5 did not ;-)

**Comment by Adminius** [comment]
Hey, any news?
I will also use Tasmota (H801 and Sonoff B1 Bulb) with my KNX installation with the dimmer functionality (long press on a button for change the brightness).
I also tried https://github.com/ascillato/Tasmota_KNX/releases/tag/v9.3.1.2
but without success:
If I send 100% DPT 5.001 via ETS (255/FF value) I get this: 
21:30:43.576 KNX: Received from 4.3.14 Command Write: null to KNX RX 1
If I send 50%:
21:31:29.520 KNX: Received from 4.3.14 Command Write: -0.00 to KNX RX 1
and with 0%:
21:31:59.322 KNX: Received from 4.3.14 Command Write: 0.00 to KNX RX 1

**Reply by ascillato** [reply]
Hi, it is not supported at this moment, sorry. I have to find some time to code it.

**Comment by tetzlav** [comment]
Its not clear and i dont find any official documentation to this:
Does Tasmota supports dimming control via knx?
And if so what datapoint are used? (relative 3-bit DP3.007 or absolute 8-bit unsigned DP5.001)

**Reply by ascillato** [reply]
Hi, it is not supported yet.

**Reply by Adminius** [reply]
@ascillato can you do it please? ;)
I made it for our knx-library, but tasmota is to complex for me
https://github.com/KONNEKTING/ALEDD1/blob/master/led_functions.h#L117-L131

dimmer -> https://github.com/Adminius/DimmerControl

**Reply by ascillato** [reply]
It is in my TO DO list. I want to add it. At this moment I don't have much free time. As soon as it is possible for me I will add it.

**Reply by tetzlav** [reply]
Can i do something to help implementing knx dimming control?

**Reply by ascillato** [reply]
It is in my TO DO list. I want to add it. At this moment I don't have much free time. As soon as it is possible for me I will add it.

**Reply by WoutervanZijl** [reply]
Is the a way to contribute to this situation?:)

**Reply by Turtle051987** [reply]
Can we expect the support on dimming in 2024?

**Reply by barbudor** [reply]
Hello @Turtle051987 

ascillato has left the team to focus on personal things so we don't have any more a KNX specialist in the team.

Although I am not using KNX myself and I'm not owning ETS, I recently tried to help and made a couple of changes in the KNX driver.

If you could express clearly what you would expect, pointing in the KNX specifications what should be done, I could have a look

**Reply by Turtle051987** [reply]
Hi @barbudor 
thank you for your reply. What i would need is a support for KNX DPT5 which is 1 Byte and 8-bit unsigned value. 
![image](https://github.com/arendst/Tasmota/assets/165657343/0fd2cd5a-85e2-47eb-a99e-b98461938692)
In the dropdown for connecting a KNX group adress to a tasmota function the option for "dimming value" must be added for sending and receiving KNX commands to connect a KNX DPT5 1 Byte and 8-bit unsigned value.
![image](https://github.com/arendst/Tasmota/assets/165657343/bf5c8355-2cde-4a85-bd7f-a2d237e32e23)

I have the hardware an infrastructure for tests so I would like to offer to support you with the tests.
I would be happy to have this feature. When it is working we can have a closer look at the RGB feature.

**Reply by Noschvie** [reply]
@Turtle051987 what is your use case?

My use case would be to forward dim + and dim - KNX messages from a MDT Glastaster to a Zigbee device connected by Tasmota zigbee / Sonoff zigbee pro.

**Reply by Turtle051987** [reply]
@Noschvie I want to control a bulp withing my KNX visualisation. Of cource the dim+ and dim- would be a nice feature but we have to keep in mind, that this feature would need a further parameter for the dimming time in the Tasmota device because the device must know which step it should dim brighter oder darker when you push the button on your Glastaster oder when you are holding the button. So that version of dimming is much more difficult. I would suggest to start with the implementation of an easier feature like the absolute dimming value and then ad more features step by step. But I'm also fine if @barbudor wants to do it all at the same time.

**Reply by barbudor** [reply]
From the doc available [here](https://support.knx.org/hc/en-us/articles/15392604906514-Interworking-Datapoint-types), I have identified some datapoint of interrest:
- DPT_Colour_RGB (232.600) = 3 x uint8 : red, green, blue 0..255
- DPT_RelativeControl_RGB (254.600) = 3 x uint8 : 3x (1bit:up/down, 3bits:step)
- DPT_Coulour_RGBW (251.600) = 6 x uint8 : red, green, blue (0..255), reserved (0), 4 bits specifying validity of red, gree, blue
- DPT_RelativeControl_RGBW (252.600) = 5 x uint8 : 4 x (1bit:up/down, 3bits:step), last byte is validity mask
- DPT_Control_Dimming (3.007) = 1 uint8 : 1 bit up/down, 3 bit step
- DPT_Scaling (5.001) = 1 uint8 0..100

Does that ring a bell ?
Do you see somewhere in your software (ETS ?) something that would confirm that ? 

What kind of light do you have ? Monochrome ? RGB ? RGBW ?

I can start with absolute Dimmer (5.001) and Colour RGB (232.600) and let's see if it works ?

What ESP model would like to test ? ESP8266 ? ESP32 ? ESP32C3 ? ...

**Reply by Turtle051987** [reply]
> DPT_Scaling (5.001) = 1 uint8 0..100
is the right for my request for absolute dimming and I can confirm that also in ETS. I have a HAMA monochrome bulp (https://templates.blakadder.com/hama_00176555.html) which has the ESP8266. I can also confirm that RGB (232.600) is a possibility to set the colour of a RGB oder RGBW device in ETS.
![image](https://github.com/arendst/Tasmota/assets/165657343/2c9dca8a-24db-4a3a-bbe3-b2e493c16400)
But I have no Tasmota hardware to test this. I like your suggestion to start with absolute Dimmer (5.001) and Colour RGB (232.600).

**Reply by barbudor** [reply]
I'm confused @Turtle051987 
You are the one asking for support but you then say you don't have the hardware 
On my side I don't have the software (ETS)

So where are we going ?

At least you should get an ESP dev board with a RGB led connected to it

**Reply by Turtle051987** [reply]
@barbudor I just need the support for dimming DPT5 and I think with the mentioned Hama bulp I have the right hardware to test if you can implement this. At the moment I don't need any other DPT for my project.
Sorry, I did not want to confuse you!

**Reply by barbudor** [reply]
Progressing ...

![image](https://github.com/arendst/Tasmota/assets/2996042/49f45ec4-0a5f-4303-9b47-0fa8be52d0b4)
![image](https://github.com/arendst/Tasmota/assets/2996042/7c0b3632-a373-4e64-8b7d-c227b973fc0e)

**Reply by fluppie** [reply]
Cool, I have a Tasmotized H801 with a "regular" RGBW led strip. Today I use a KNX smart switch to put it on/ff. Would be cool to change intensity as well from the switch :-). Following this.

**Reply by Noschvie** [reply]
> support for dimming DPT5

Which device is sending this data point / absolute dimming value ?

**Reply by Turtle051987** [reply]
I'm using the visualisation from EDOMI and sometimes the Gira G1.

**Reply by Jaegerstefan** [reply]
H801 here too. Would be great to use the dimming feature with some MDT KNX devices.

**Reply by fluppie** [reply]
Here also MDT switches 
![csm_BE-GT2TW-02_EN_D_Heating_F_288b052b2e](https://github.com/arendst/Tasmota/assets/8402106/dfb0c65f-8148-433c-b982-c8038c04f47e)

**Reply by barbudor** [reply]
Guys, here is a first version which supports only Absolute values

```
Light type | Dimmer (Bright)   | Color                     |
-----------|-------------------|---------------------------|
Single     | DPT_Scaling 5.001 | not supported +           |
CT         | DPT_Scaling 5.001 | not supported  +          |
RGB        | DPT_Scaling 5.001 | DPT_Colour_RGB 232.600    |
RGBW       | DPT_Scaling 5.001 | DPT_Colour_RGBW 251.600 * |
RGBCW      | DPT_Scaling 5.001 | not supported  +          |

* the validity bits are not supported. Tasmota expect all 4 values R, G, B, W to be present
+ those are not supported because I haven't found a matching DPT for 2 and 5 channels.
```

In KNX configuration menu, `Bright` and `Color` appears only is you have a LIGHT configured which includes PWM channels or WS2812 or any similar controller

You must be sure that your KNX software use the proper DPT

Using Wireshalr and NodeRED+KNX Ultimate I have tested:
- Sending KNX commands (WRITE) to Tasmota
- Receiving response to those commands
- Receiving KNX updates from tasmota when the light is chnged through UI or console

I have not been able to test the request/answer mode when a host wants in inquire about current status

Let's see how this works, if it needs some adjustment
If needed we can look into moving further with relative commands

Here is an ESP8266 test binary: [tasmota-knx-TEST.zip](https://github.com/arendst/Tasmota/files/14877456/tasmota-knx-TEST.zip)

If anyone wants to compile a custom biuld, my branch is there: https://github.com/barbudor/Tasmota/tree/tas11733_KNX_dimmer_and_rgb_contol

I won't be able to make changes and compile new binaries before saturday or sunday evening

**Reply by barbudor** [reply]
@Jaegerstefan @fluppie To implement new modes I need to understand which DPT they are sending in order to build a relation between a DPT and a Tasmota command

**Reply by Turtle051987** [reply]
@barbudor I testet the absolute dimming DPT5.001 and it is working as I expected. I can control the bulp through ETS and it also sends the right state back to KNX when changing in the webinterface.
![image](https://github.com/arendst/Tasmota/assets/165657343/8630fc56-91e7-4d71-885c-33ea1cbc5e07)
Thank you so much!

**Reply by barbudor** [reply]
Thanks
Let's have the other guys testing RGB and I could PR that first version
Then we could look at other modes like relative

I also may have found some DPT that could work for temperature but that would need some confirmation from people more experienced in KNX

**Reply by eldi0s** [reply]
Thanks for your effort here, really appreciated. I was dealing with this on HomeAssistant to control a Magic Home RGBW controller wiht a MDT Glastaster via KNX. Heres a blueprint I found and used as a starting point.

[KNX realtive dimming HA blueprint](https://gist.github.com/kalhimeo/88f6e38b45aa872961cc7b748eacef3d)

This blueprint is showing how realtive dimming could work. it also supports a helper to implement switch on while dim up from 0 or switch off at dim down to 0. Works well in my setup btw. All assuming, that the white channel out of the RGBW controller is treated as a seperate light entity with brightness control.

I was also trying to implement RGB relative dimming in the same way via the Glastaster and was working on conversions from HSV2RGBW and RGBW2HSV via some scripts called from the blueprint. The Glastaster can only deal with HSV (Hue) values 
[H=0-360°, S=0-100%, V=0-1] and the MagicHome Controller in HomeAssistant uses individual RGB and a brightness value [0-255,0-255,0-255 and 0-100%). I did only get it to work one way, so HomeAssistant communicates absolute changes in RGB and brightness back to the Glastaster as HSV. Didn't find the time to implement relative RGB changes from the Glastaster to the MagicHome Controller as this is a bit more delicate.

I can provide more insight if it supports your work. Let me know what you need. Would love to flash this MagicHome with Tasmota and use the KNX RGBW functions with my Glastaster.

THX

**Reply by Turtle051987** [reply]
Hi @barbudor my further tests for the dimming were good. Just one point that is currently not like most KNX devices are reacting: when the light is switched off the bulp shpuld send 0% for bringhtness and the new/old brightness when it is switched on again. Do you think it's possible to add this?

**Reply by barbudor** [reply]
HI @Turtle051987 
Good to know its working so far
I will look into what you are asking but that would be a different behaviour than standard Tasmota behavior (such as MQTT reporting for the same)

**Reply by Turtle051987** [reply]
Hi, it seems like the changes have not been implemented for Version 14. As the absolute dimming is working I want to ask if it's possible to add this festure for the next release.

**Reply by arendst** [reply]
Try latest dev.

**Reply by barbudor** [reply]
@Turtle051987
Sorry I was late to push the updated code to Tasmota
I did that only yesterday so it missed the new release. Still available in dev as pointed by Theo

**Reply by Laudi111** [reply]
@barbudor: At first, thank you for the KNX dimming extension.

I'm current testing KNX dimming with Tasmota Release 14.2 and i had some problems.
I use a Shelly RGBW2 in configuration "white Option (setoption68)" with 4 white LED strips and i want to control the brightness of every LED strip. 

When i change the Brightness over KNX, the "Bright" value changes in Tasmota console view, but it doesn't take effect on the brightness slider on any one of the 4 channels. I think its seems to be a problem because of 4 channels and not RGBW configuration. Is it possible to get a knx brightness value for every channel instead of color?
![1](https://github.com/user-attachments/assets/a50701d6-73b8-43a3-9faa-09c0d9c734bc)
![2](https://github.com/user-attachments/assets/c7ddb7f4-682b-42a2-8482-2964f63f0fed)

If i change the "Color" over KNX, the 4 channels become updated with the "color" value. 
But it looks like a problem with the 4th "W"-channel, it's limited to the value 64 when receiving a value over KNX. In web-interface i can slide it to 255 and the value is also sent back to KNX. But when i'm sending $FF on all 4 channel over KNX, the first 3 channels are set to maximum but the fourth ist set to $3F. Can you check this behavior?

![3](https://github.com/user-attachments/assets/ef33462e-c447-4972-be43-9746380375fe)
![4](https://github.com/user-attachments/assets/3d507848-24d9-463b-a081-ecd7f2bba34b)
![5](https://github.com/user-attachments/assets/de791b77-aa6a-49a0-ba0b-07430aaa7432)

**Reply by barbudor** [reply]
@Laudi111 the KNX implementation is meant for Light mode of Tasmota, not separated PWM channels
You might be luckier using the TX/RX modes with rules
But I'm not a KNX user myself, just did some coding to help

**Reply by digit666** [reply]
@Laudi111 Have you get the dimmer function of RGBW2 (WHITE) running with Tasmota and KNX? I´ve the same problem and it would be nice if I can dim the four LED WHITE channels by KNX. Or have you choose the solution with Rules?

**Reply by Hora0816** [reply]
Hi, 
is there already any solution on this issue? I'm using 4 PWM in the RGBW config to control a LEDVANCE RGB Wifi spot (replaced the BK chip with EPS8266). It look to me as the ESP is receiving more than 6 bytes from the knx bus - in the ETS monitor I see 6 bytes sent:   x00000020000f (DTP251.600) to esp and x00000050000f received from esp while and the tasmota log says : 13:10:51.290 KNX: Received from 3/1/70 Command Write: 00000040103758 to Color
13:10:51.298 KNX: Color 0,0,0,80 sent to 3/1/71.  
looks to me as KNX-RX telegram mapping issues ( wifi reception issues  I would exclude as the  esp is about in 1m distance to the wifi AP), the reception from ESP to KNX is ok.

**Reply by Laudi111** [reply]
I'm now using rules. It's not perfect but for my usecase it's still ok.

**Comment by Hora0816** [comment]
...I used rules and knxTx/knxRX  with plain RGBW values but the White values is always wrong and the dtp 251.600 reply shows a fixed withe value. Using knx 251.600 as command at reception shows more than 6 bytes corrupting the white value.With MQTT or http commands no problems.Von meinem/meiner Galaxy gesendet
-------- Ursprüngliche Nachricht --------Von: Laudi111 ***@***.***> Datum: 10.11.25  18:34  (GMT+01:00) An: arendst/Tasmota ***@***.***> Cc: Hora0816 ***@***.***>, Comment ***@***.***> Betreff: Re: [arendst/Tasmota] KNX: Add RGB and Dimming Control (Discussion #11733) 
I'm now using rules. It's not perfect but for my usecase it's still ok.
